packages/nimble-components/src/table/specs/table-column-specs/table-column-duration-text-hld.md

```diff
@@ -0,0 +1,227 @@
+| "0 sec"                | "0 Sek."                  | "0 s"              | "0 秒"                  |
+
+#### `DurationFormatter`
+
+We will export the class that is responsible for formatting the duration values so that we can use it for formatting values outside of the `Table` (a requirement in SLE).
+
+```ts
+export class DurationFormatter {
+    public constructor(private readonly lang: string) {}
+
+    // formats a value in the seconds domain into a localized display value that can
+    // consist of days, hours, minutes, and seconds.
+    public format(value: number): string {}
+}
+```
+
+### Angular integration
+
+An Angular directive will be created for the component. The component will not have form association, so a `ControlValueAccessor` will not be created. The `DurationFormatter` will be re-exported from Angular from a new `formatters` entrypoint.
+
+We will also export a `DurationPipe` that will use the `DurationFormatter` under the hood.
+
+### Blazor integration
+
+A Blazor wrapper will be created for the component. Blazor will not provide a means to access the formatting function of the `DurationFormatter` until there is a client requirement for it.
+
+### Visual Appearance
+
+Will use same text styling as the `nimble-table-column-text`.
+
+## Implementation
+
+In order to format a single value into multiple parts with localized text, we can provide a similar implementation to the current [`ElapsedTimePipe` `transform` method](https://dev.azure.com/ni/DevCentral/_git/Skyline?path=/Web/Workspaces/SystemLinkShared/projects/systemlink-lib-angular/src/pipes/elapsed-time.pipe.ts&version=GBmaster&line=18&lineEnd=19&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) used in SLE, but instead of using the Angular `DecimalPipe` to format the value, we can just use the native `Intl.NumberFormat`, and pass in the locale we get from the `lang` setting on the `nimble-theme-provider`:
+
+```ts
+const result = [];
+const hours = Math.floor(value / 3600);
+if (hours) {
+    const hoursFormatter = new Intl.NumberFormat(this.lang, { style: 'unit', unit: 'hour' });
+    const formattedHours = hoursFormatter.format(hours);
+    result.push(formattedHours);
+}
+... // do similar work for minutes and seconds
+```
+
+Other implementation details will conform to existing patterns all other table columns handle, such as `CellView` and `GroupHeaderView` implementations.
+
+_Note: For number values that can't be translated into a representable time (e.g. `NaN`, `∞`, `-∞`, negative values), we will simply show a blank cell._
+
+### Considered Corner Cases
+
+#### Really large numbers
+
+At some point numbers that can be formatted as a duration become too large to be practically represented in such a way. We will have a cutoff value (in days) where when the value exceeds that limit, we will revert to displaying the duration as a scientific representation of the value in the seconds unit.
```

Comment:
For very large numbers it seemed more intuitive than saying 3.264E61 days
Curious about the concern since I just recommended the same thing to @m-akinc for very large and very small numbers we don't have localized prefixes for.

Reply:
I don't have a strong opinion here, but I'll leave it as is until I can be swayed in the other direction.